q2_types/feature_data/_format.py

```diff
@@ -100,34 +100,53 @@ class TSVTaxonomyFormat(model.TextFileFormat):
     """
     HEADER = ['Feature ID', 'Taxon']
 
-    def sniff(self):
+    def _check_n_records(self, n=None):
         with self.open() as fh:
             data_lines = 0
             header = None
-            while data_lines < 10:
-                line = fh.readline()
+
+            file_ = enumerate(fh) if n is None else zip(range(n), fh)
+
+            for i, line in file_:
+                i = i + 1
 
                 if line == '':
                     # EOF
                     break
-                elif line.lstrip(' ') == '\n':
-                    # Blank line
+                elif line.strip(' ') == '\n':
                     continue
                 elif line.startswith('#'):
                     # Comment line
                     continue
 
-                cells = line.rstrip('\n').split('\t')
+                cells = line.strip('\n').split('\t')
+
                 if header is None:
                     if cells[:2] != self.HEADER:
-                        return False
+                        raise ValidationError("['Feature ID' and 'Taxon'] "
+                                              "must be the first two header "
+                                              "values to be a valid taxonomy "
+                                              "file.\n\nThe first two header "
+                                              "values provided are: {}."
+                                              .format(cells[:2]))
                     header = cells
                 else:
                     if len(cells) != len(header):
-                        return False
+                        raise ValidationError('Number of headers are not the '
+                                              'same as number of columns in '
+                                              'the file. \nHeader values: '
+                                              '{} \nColumn values: {} '
+                                              .format(header, cells[:], i))
+
                     data_lines += 1
 
-            return header is not None and data_lines > 0
+            if data_lines == 0:
+                raise ValidationError("No feature records found in manifest, "
```

Comment:
This isn't a manifest file, and we no longer support comments in this format (as of #219). Please update the error accordingly.

Reply:
Thanks for the feedback. I think I will also need to edit the multi-line comment at the beginning of the class, which also mentions support for comments.